lib/coins.py

```diff
@@ -293,6 +293,43 @@ def electrum_header(cls, header, height):
     def deserializer(cls):
         return Deserializer
 
+class CoinAuxPow(Coin):
+    # Set NAME and NET to avoid exception in Coin::lookup_coin_class
+    NAME = ""
+    NET = ""
+    STATIC_BLOCK_HEADERS = False
+    VERSION_AUXPOW = (1 << 8)
+
+    @classmethod
+    def header_hash(cls, header):
+        '''Given a header return hash'''
+        return double_sha256(header[:cls.BASIC_HEADER_SIZE])
+
+    @classmethod
+    def block_header(cls, block, height):
+        '''Return the AuxPow block header bytes'''
+        block = Deserializer(block)
+        start = block.cursor
+        version = block._read_le_uint32()
+        if version & cls.VERSION_AUXPOW:
+            # We are going to calculate the block size then read it as bytes
+            block.cursor = start
+            block.cursor += cls.BASIC_HEADER_SIZE # Block normal header
+            block.read_tx() # AuxPow transaction
+            block.cursor += 32 # Parent block hash
+            merkle_size = block._read_varint()
+            block.cursor += 32 * merkle_size # Merkle branch
+            block.cursor += 4 # Index
+            merkle_size = block._read_varint()
+            block.cursor += 32 * merkle_size # Chain merkle branch
+            block.cursor += 4 # Chain index
+            block.cursor += 80 # Parent block header
+            header_end = block.cursor
+        else:
+            header_end = cls.BASIC_HEADER_SIZE
+        block.cursor = start
+        return block._read_nbytes(header_end)
+
```

Comment:
One more thing, please put in a commit message the github text "Closes #xxx" for the 2 outstanding issues related to this aux pow support.  I assume it does resolve them.